source/simulator/helper_functions.cc

```diff
@@ -1320,6 +1323,104 @@ namespace aspect
+
+        AssertThrow(std::find(heating_models.begin(), heating_models.end(), "adiabatic heating") == heating_models.end(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The BA formulation expects adiabatic heating to be disabled, "
+                               "but the heating model manager found a heating plugin "
+                               "named 'adiabatic heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+
+        AssertThrow(std::find(heating_models.begin(), heating_models.end(), "shear heating") == heating_models.end(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The BA formulation expects shear heating to be disabled, "
+                               "but the heating model manager found a heating plugin "
+                               "named 'shear heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+      }
+    else if (parameters.combined_formulation == Parameters<dim>::CombinedFormulation::ALA)
+      {
+        const std::vector<std::string> &heating_models = heating_model_manager.get_active_heating_model_names();
+
+        AssertThrow(std::find(heating_models.begin(), heating_models.end(), "adiabatic heating") != heating_models.end(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The ALA formulation expects adiabatic heating to be enabled, "
+                               "but the heating model manager does not find a heating plugin "
+                               "named 'adiabatic heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+
+        AssertThrow(std::find(heating_models.begin(), heating_models.end(), "shear heating") != heating_models.end(),
+                    ExcMessage("ASPECT detected an inconsistency in the provided input file. "
+                               "The ALA formulation expects shear heating to be enabled, "
+                               "but the heating model manager does not find a heating plugin "
+                               "named 'shear heating' in the list of active plugins. "
+                               "Please check the consistency of your input file."));
+
+        const std::list<std_cxx11::shared_ptr<HeatingModel::Interface<dim> > > &heating_plugins =
+          heating_model_manager.get_active_heating_models();
+
+        for (typename std::list<std_cxx11::shared_ptr<HeatingModel::Interface<dim> > >::const_iterator plugin =
+               heating_plugins.begin(); plugin != heating_plugins.end(); ++plugin)
+          {
+            const HeatingModel::AdiabaticHeating<dim> *
```

Comment:
This would also make for a nice helper function in ``heating_model_manager``, for example ``HeatingModel::AdiabaticHeating<dim> * = heating_model_manager.get_adiabatic_heating_plugin()``.